packages/ragbits-core/src/ragbits/core/vector_stores/in_memory.py

```diff
@@ -88,7 +90,21 @@ async def retrieve(
             results: list[VectorStoreResult] = []
 
             for entry_id, vector in self._embeddings.items():
-                score = float(np.linalg.norm(np.array(vector) - np.array(query_vector))) * -1
+                if isinstance(vector, SparseVector) and isinstance(query_vector, list):
```

Comment:
Most of those cases are impossible - the embedder is configured on the vector store level, so all elements (including the query) will be embedded with the same embedder - so either both will be dense or both will be sparse. Most of this code can be deleted.